trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/ThrowableMapper.java

```diff
@@ -0,0 +1,89 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.rest;
+
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.google.inject.Inject;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.log.Logger;
+import io.trino.aws.proxy.spi.rest.Request;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.container.ResourceContext;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.ResponseBuilder;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import org.glassfish.jersey.server.ContainerRequest;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static jakarta.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static jakarta.ws.rs.core.MediaType.APPLICATION_XML_TYPE;
+import static java.util.Objects.requireNonNull;
+
+public class ThrowableMapper
+        implements ExceptionMapper<Throwable>
+{
+    private static final Logger log = Logger.get(ThrowableMapper.class);
+    private static final String X_AMZ_REQUEST_ID = "x-amz-request-id";
+
+    @Context
+    private ResourceContext resourceContext;
+
+    private final XmlMapper xmlMapper;
+
+    @Inject
+    public ThrowableMapper(XmlMapper xmlMapper)
+    {
+        this.xmlMapper = requireNonNull(xmlMapper, "xmlMapper is null");
+    }
+
+    @Override
+    public Response toResponse(Throwable throwable)
+    {
+        ContainerRequest containerRequest = resourceContext.getResource(ContainerRequest.class);
+        Optional<String> requestId = Optional.ofNullable((Request) containerRequest.getProperty(Request.class.getName()))
+                .map(Request::requestId)
+                .map(UUID::toString);
+
+        HttpStatus status = switch (throwable) {
+            case WebApplicationException webApplicationException -> HttpStatus.fromStatusCode(webApplicationException.getResponse().getStatus());
+            default -> HttpStatus.INTERNAL_SERVER_ERROR;
+        };
+
+        try {
+            log.error(throwable, "Request failed for %s", containerRequest.getRequestUri().getPath());
+
+            ErrorResponse response = requestId
+                    .map(id -> new ErrorResponse(
+                                status.reason(),
+                                throwable.getMessage(),
+                                containerRequest.getRequestUri().getPath(),
+                                id))
+                    .orElse(new ErrorResponse(
+                            status.reason(),
+                            throwable.getMessage(),
+                            containerRequest.getRequestUri().getPath()));
```

Comment:
No strong opinion, but would it be easier to just do `requestId.orElse(null)`?

`ErrorResponse` already deals with null request IDs. It doesn't feel as elegant, but we're duplicating the construction of this record just to be able to map the request ID optional

Reply:
We can make the `requestId` an optional, but then it will be serialised to an empty tag:

```xml
<Error><Code>a</Code><Message>b</Message><Resource>c</Resource><RequestId/></Error>
```

This seems to work and we avoid nulls. wdyt?